src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -0,0 +1,729 @@
+                MoveDown(_nodes[index], index);
+            }
+        }
+
+        /// <summary>
+        /// Moves a node up in the tree to restore heap order.
+        /// </summary>
+        private void MoveUp((TElement element, TPriority priority) node, int nodeIndex)
+        {
+            // Instead of swapping items all the way to the root, we will perform
+            // a similar optimization as in the insertion sort.
+
+            while (nodeIndex > 0)
+            {
+                int parentIndex = GetParentIndex(nodeIndex);
+                var parent = _nodes[parentIndex];
+
+                if (Comparer.Compare(node.priority, parent.Priority) < 0)
+                {
+                    _nodes[nodeIndex] = parent;
+                    nodeIndex = parentIndex;
+                }
+                else
+                {
+                    break;
+                }
+            }
+
+            _nodes[nodeIndex] = node;
+        }
+
+        /// <summary>
+        /// Moves a node down in the tree to restore heap order.
+        /// </summary>
+        private void MoveDown((TElement element, TPriority priority) node, int nodeIndex)
+        {
+            // The node to move down will not actually be swapped every time.
+            // Rather, values on the affected path will be moved up, thus leaving a free spot
+            // for this value to drop in. Similar optimization as in the insertion sort.
+
+            int i;
+            while ((i = GetFirstChildIndex(nodeIndex)) < _size)
+            {
+                // Check if the current node (pointed by 'nodeIndex') should really be extracted
+                // first, or maybe one of its children should be extracted earlier.
+                var topChild = _nodes[i];
+                int childrenIndexesLimit = Math.Min(i + Arity, _size);
+                int topChildIndex = i;
+
+                while (++i < childrenIndexesLimit)
+                {
+                    var child = _nodes[i];
+                    if (Comparer.Compare(child.Priority, topChild.Priority) < 0)
+                    {
+                        topChild = child;
+                        topChildIndex = i;
+                    }
+                }
+
+                // In case no child needs to be extracted earlier than the current node,
+                // there is nothing more to do - the right spot was found.
+                if (Comparer.Compare(node.priority, topChild.Priority) <= 0)
+                {
+                    break;
+                }
+
+                // Move the top child up by one node and now investigate the
+                // node that was considered to be the top child (recursive).
+                _nodes[nodeIndex] = topChild;
+                nodeIndex = topChildIndex;
+            }
+
+            _nodes[nodeIndex] = node;
+        }
+
+        public partial class UnorderedItemsCollection : IReadOnlyCollection<(TElement element, TPriority priority)>, ICollection
```

Comment:
```suggestion
        public sealed class UnorderedItemsCollection : IReadOnlyCollection<(TElement element, TPriority priority)>, ICollection
```

Reply:
Hmm...
a) In the reference file above you suggested `public partial sealed class`
b) Here you suggested `public sealed class`

Do you wish this to be (a) or (b)?

Reply:
> Do you wish this to be (a) or (b)?

Yes ;-)

`public partial sealed class` in the ref and `public sealed class` here.  The `partial` has no actual impact on the signature once compiled, and is an artifact of how our ref files are authored (and automatically generated).